pkg/ui/app/containers/timescale.tsx

```diff
@@ -29,46 +66,61 @@ interface TimeScaleSelectorProps {
   defaultTimescaleSet: boolean;
 }
 
-interface TimeScaleSelectorState {
-  controlsVisible: boolean; // Timescale selector controls visible.
-}
-
-class TimeScaleSelector extends React.Component<TimeScaleSelectorProps, TimeScaleSelectorState> {
-
-  timescaleBtn: Element;
-
-  constructor() {
-    super();
-    this.state = {
-      controlsVisible: false,
-    };
+// TimeScaleSelector is the dropdown that allows users to select the time range
+// for graphs.
+class TimeScaleSelector extends React.Component<TimeScaleSelectorProps, {}> {
+  changeSettings = (newTimescaleKey: SelectorOption) => {
+    let newSettings = timewindow.availableTimeScales[newTimescaleKey.value];
+    if (newSettings) {
+      this.props.setTimeScale(newSettings);
+    }
   }
 
-  setVisible = (visible: boolean) => {
-    this.setState({
-      controlsVisible: visible,
-    });
-  };
+  arrowClick = (direction: ArrowDirection) => {
+    let selected = _.clone(this.props.currentScale);
+
+    switch (direction) {
+      case ArrowDirection.RIGHT:
+        if (selected.windowEnd) {
+          selected.windowEnd.add(selected.windowSize);
+        }
+        break;
+      case ArrowDirection.LEFT:
+        selected.windowEnd = selected.windowEnd || moment();
+        selected.windowEnd.subtract(selected.windowSize);
+        break;
+      default:
+        console.error("Unknown direction: ", direction);
+    }
 
-  hide = (e: Event) => {
-    if (e.target !== this.timescaleBtn) {
-      this.setVisible(false);
+    if (!selected.windowEnd || selected.windowEnd > moment().subtract(selected.windowValid)) {
```

Comment:
Please add a comment here about what this is doing (making the time window auto-update again).